src/Repository/DatasetRepository.php

```diff
@@ -338,4 +338,25 @@ public function getDatasetWithDoiSet(): array
             ->getResult()
         ;
     }
+
+    /**
+     * Get all datasets, filter aware
+     *
+     * @return Dataset[]
+     */
+    public function findAllFiltered(): array
+    {
+        $qb = $this->createQueryBuilder('dataset');
+        $qb->select('dataset');
+
+        if ($this->fundingOrgFilter->isActive()) {
+            $researchGroupIds = $this->fundingOrgFilter->getResearchGroupsIdArray();
+
+            $qb
+            ->innerJoin('dataset.researchGroup', 'rg')
+            ->andWhere('rg.id IN (:rgs)')
+            ->setParameter('rgs', $researchGroupIds);
+        }
+        return $qb->getQuery()->getResult();
+    }
```

Comment:
This is not needed, `FundingOrgFilter` has a function that will return a list of ID's, and then you can just do Repo->findBy(arrayOfIds)

Example: https://github.com/griidc/pelagos/blob/develop/src/Controller/DefaultController.php#L82